Documentation Fix: Corrects explanation of what happens when dependent is not set

alex-handley opened the pull request:
I was reading over the documentation for dependent and found a mistake. 
The default behaviour when dependent is not set is the foreign key is left unchanged after destroy. The documentation says nullify is the default which is incorrect.

The following merge/issue fixed the same mistake higher up in the documentation #5828 #7188

Thanks,
Alex

bansalakhil commented:
:+1:

I was planning to raise a PR for the same, just a few mins back.

alex-handley commented:
I finally finished a blog post on this at the weekend :) - http://blog.shoreditchclimber.co.uk/rails-dependent-associations/

vipulnsward commented on activerecord/lib/active_record/associations.rb:
```diff
@@ -1014,7 +1014,7 @@ def association_instance_set(name, association)
     # do the deletion according to the strategy specified by the <tt>:dependent</tt> option, or
     # if no <tt>:dependent</tt> option is given, then it will follow the default strategy.
-    # The default strategy is <tt>:nullify</tt> (set the foreign keys to <tt>nil</tt>), except for
+    # The default strategy is to do nothing (leave the foreign keys with the parents id set), except for
```
`parent's`

vipulnsward commented on activerecord/lib/active_record/associations.rb:
```diff
@@ -1001,7 +1001,7 @@ def association_instance_set(name, association)
     #
     # Note that <tt>:dependent</tt> option is ignored for +has_one+ <tt>:through</tt> associations.
-    # 
+    #
```
unnecessary change here.

vipulnsward commented:
@alex-handley In future for doc fixes, please add `[ci skip]` in commit message, so as to not run tests on travis.

alex-handley committed:
[ci skip] corrects documentation for the default dependent behaviour

By default the foreign key will remain set with the parent id after
destroy is fired.

alex-handley commented:
@vipulnsward Thanks for the advice! This is my first PR to rails.  
I have made the changes you suggested.
